### PR TITLE
show posts in ag grid

### PR DESCRIPTION
**Pull request by aboubakar20**

### CODE REVIEW

File: src/components/PostsTable.vue [hunk 1]
```diff
@@ -1,104 +1,79 @@
...
+  data() {
+    return {
+      columnDefs: null,
+      rowData: null,
+      frameworkComponents: null,
+    };
+  },
   props: ["posts"],
+  methods: {
+    onGridReady(params) {
```
Comment (junaidrt) [inline src/components/PostsTable.vue]: Please remove unused methods.

File: src/components/btn-cell-renderer.vue [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<template>
```
Comment (junaidrt) [inline src/components/btn-cell-renderer.vue]: Please change name of component to PascalCase.

File: src/main.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 import Vue from 'vue'
 import App from './App.vue'
 
+import "../node_modules/ag-grid-community/dist/styles/ag-grid.css";
```
Comment (junaidrt) [inline src/main.js]: Please move these css imports to local components.

**Review by junaidrt — CHANGES_REQUESTED** [review]

File: src/components/PostsTable.vue [hunk 4]
```diff
@@ -1,104 +1,79 @@
...
+      this.gridApi = params.api;
+      this.columnApi = params.columnApi;
+    },
+  },
+  beforeMount() {
+    this.rowData = this.posts;
+    this.columnDefs = [
```
Comment (junaidrt) [inline src/components/PostsTable.vue]: Can we move this initialization into another function and call the function in `data` property of vue?

**Review by junaidrt — CHANGES_REQUESTED** [review]

File: src/components/PostsTable.vue (hunk 1 above)
Comment (aboubakar20) [reply src/components/PostsTable.vue]: Removed unused methods.

File: src/components/btn-cell-renderer.vue (hunk 2 above)
Comment (aboubakar20) [reply src/components/btn-cell-renderer.vue]: Changed

File: src/main.js (hunk 3 above)
Comment (aboubakar20) [reply src/main.js]: CSS imports moved to its local component

File: src/components/PostsTable.vue (hunk 4 above)
Comment (aboubakar20) [reply src/components/PostsTable.vue]: Done

**Pull request merged**